src/Auth/Jwt.ts

```diff
@@ -119,20 +115,40 @@ export class Jwt implements IAccessToken {
 	 * @param {IJwtBody} body
 	 * @param {string} signature
 	 */
-	constructor (
-		public readonly header: IJwtHeader,
-		public readonly body  : IJwtBody,
-		public readonly signature?: string
-	) {
-		const withoutSignature = this.headerBase64() + '.' + this.bodyBase64();
-
-		this.unsignedData = withoutSignature;
-
-		if (this.signature == null) {
-			this.stringRepresentation = withoutSignature;
+	constructor (header: IJwtHeader, body  : IJwtBody, signature?: string);
```

Comment:
I thinking about notation like that (considering https://www.typescriptlang.org/docs/handbook/functions.html#Overloads):
```
	constructor (header: IJwtHeader, body  : IJwtBody, signature?: string);
	constructor (jwtString: string);
	constructor (header: any, body?: any, signature?: any) {
```
first two lines defines the wanted declarations and third the all possible cases (`any` because of JavaScript usage). I don't sure do we need use any for that case, but I want try it, because it will help to catch some illegal inputs from javascript